desc/equilibrium/equilibrium.py

```diff
@@ -237,12 +261,12 @@ def __init__(
         _assert_nonnegint(N_grid, "N_grid")
 
         self._N = int(setdefault(N, self.surface.N))
-        self._M = int(setdefault(M, self.surface.M))
+        self._M = int(setdefault(M, setdefault(self.surface.M, self.xsection.M)))
```

Comment:
This might give weird results for the reasons above. IE, if the user passes in a xsection with M=10, then self.surface will be the default surface with M=1, so eq.M will be 1 not 10